src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -221,24 +263,176 @@ public void onTraversalStart() {
      */
     @Override
     public void traverse() {
-        try {
-            importer.importBatch();
-        } catch (IOException e) {
-            throw new UserException("GenomicsDB importBatch raised exception", e);
+
+        int updatedBatchSize = (batchSize == DEFAULT_ZERO_BATCH_SIZE) ? sampleNames.size() : batchSize;
+        int sampleCount = sampleNames.size();
+        int totalBatchCount = (sampleCount/updatedBatchSize) + (sampleCount%updatedBatchSize==0 ? 0 : 1);
+
+        GenomicsDBImporter importer;
+
+        for (int i = 0, batchCount = 1; i < sampleCount; i += updatedBatchSize, ++batchCount) {
+
+            Map<String, FeatureReader<VariantContext>> sampleToReaderMap =
+                getFeatureReaders(sampleNames, sampleNameToVcfUri, updatedBatchSize, sampleCount, i);
+
+            logger.info("Importing batch " + batchCount + " with " + sampleToReaderMap.size() + " samples");
+            final long variantContextBufferSize = vcfBufferSizePerSample * sampleToReaderMap.size();
+
+            GenomicsDBImportConfiguration.ImportConfiguration importConfiguration =
+                    createImportConfiguration(workspace, arrayName,
+                            variantContextBufferSize, segmentSize,
+                            "", "",
+                            i, (i+updatedBatchSize-1));
+
+            try {
+                importer = new GenomicsDBImporter(sampleToReaderMap, mergedHeader, intervals.get(0), importConfiguration);
+            } catch (IOException e) {
+                throw new UserException("Error initializing GenomicsDBImporter in batch " + batchCount, e);
+            }
+
+            try {
+                importer.importBatch();
+                assert importer.isDone();
+            } catch (IOException e) {
+                throw new UserException("GenomicsDB import failed in batch " + batchCount, e);
+            }
+
+            closeReaders(sampleToReaderMap);
+            logger.info("Done importing batch " + batchCount + "/" + totalBatchCount);
         }
     }
 
     @Override
     public Object onTraversalSuccess() {
-        logger.info("Import completed!");
-        return importer.isDone();
+        if (batchSize==DEFAULT_ZERO_BATCH_SIZE) {
+            logger.info("Import completed!");
+        } else {
+            logger.info("Import of all batches to GenomicsDB completed!");
+        }
+
+        // Write the vid and callset map JSON files
+        GenomicsDBImporter.writeVidMapJSONFile(vidMapJSONFile.getAbsolutePath(), mergedHeader);
+        GenomicsDBImporter.writeCallsetMapJSONFile(callsetMapJSONFile.getAbsolutePath(), callsetMappingPB);
```

Comment:
Yeah, these writes throw FileNotFoundException which is now propagated to this tool and thrown as an UserException